pkg/store/consul/auditlogstore/consul_store.go

```diff
@@ -0,0 +1,128 @@
+package auditlogstore
+
+import (
+	"encoding/json"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/square/p2/pkg/util"
+
+	"github.com/hashicorp/consul/api"
+	"github.com/pborman/uuid"
+)
+
+const auditLogTree string = "audit_logs"
+
+// ConsulStore is responsible for storage and retrieval of AuditLog JSON
+// objects. It operates a little differently than the other consul store types
+// in this project because the creation of an audit record is always expected
+// to be a part of a transaction. As a result, it doesn't actually need access
+// to a consul client to create records, it will simply save them to a
+// transaction object.
+type ConsulStore struct {
+	consulKV ConsulKV
+}
+
+type ConsulKV interface {
+	List(prefix string, opts *api.QueryOptions) (api.KVPairs, *api.QueryMeta, error)
+}
+
+func NewConsulStore(consulKV ConsulKV) ConsulStore {
+	return ConsulStore{
+		consulKV: consulKV,
+	}
+}
+
+func (ConsulStore) Create(
+	txn *api.KVTxnOps,
```

Comment:
i'm eager for feedback on this way of doing things. The workflow will look sort of like a mysql transaction: you start a transaction, apply operations on it, and then commit. Of course in this case it's all local rather than on the server